pkg/util/mount/nsenter_mount_test.go

```diff
@@ -72,120 +75,668 @@ func TestParseFindMnt(t *testing.T) {
 	}
 }
 
-func TestGetPidOnHost(t *testing.T) {
-	tempDir, err := ioutil.TempDir("", "get_pid_on_host_tests")
+func TestCheckDeviceInode(t *testing.T) {
+	// First directory is always on $TEMP, typically on /tmp
+	fs1Dir, err := ioutil.TempDir("", "nsenter-mounter-device-")
+	if err != nil {
+		t.Fatalf("Cannot create temporary directory: %s", err)
+	}
+	defer os.RemoveAll(fs1Dir)
+
+	// Find a place where an unit test can write and is on a different device
+	// than $TEMP. Skip the test if there is no such place.
+	var fs1Stat, fs2Stat unix.Stat_t
+	var fs2Dir string
+
+	err = unix.Stat(fs1Dir, &fs1Stat)
 	if err != nil {
-		t.Fatalf(err.Error())
+		t.Fatalf("Cannot stat temporary directory: %s", err)
+	}
+
+	candidates := []string{
+		// Fedora, RHEL/CentOS and COS have /tmp on tmpfs and /var/tmp on a disk
+		"/var/tmp",
+		// Use /dev/shm as the last resort. It has different purpose than /tmp,
+		// but it's usually world-writable tmpfs and we clean up our mess
+		// quickly. Let's hope nobody notices.
+		"/dev/shm",
```

Comment:
yeahhhhh I don't know about /dev/shm...  Maybe just /tmp or /var/tmp? 

You can still run the 2 "same fs" test cases on any system.  It's only the 3rd one that may not be able to run everywhere.

Reply:
Ha! I don't need two writable directories, one /tmp is enough. I can test with already existing file that's for sure on a different device, such as /proc/self/status - it guaranteed to exists, be readable and on a different "device" than /tmp.